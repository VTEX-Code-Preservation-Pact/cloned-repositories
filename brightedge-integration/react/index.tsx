import React, { FunctionComponent, useState, useEffect } from 'react'

import useScript from './hooks/useScriptLoader'

interface Props {
  appSettings: any
}

const BrightEdgeScript: FunctionComponent<{ accountId: string }> = ({
  accountId,
}) => {
  const [loaded, error] = useScript(
    `//cdn.bc0a.com/autopilot/${accountId}/autopilot_sdk.js`
  )

  if (!loaded || error) return null
  return null
}

const BrightEdge: FunctionComponent<Props> = ({ appSettings }) => {
  const { accountId } = appSettings
  const [shouldLoadScript, setShouldLoadScript] = useState(false)

  useEffect(() => {
    setShouldLoadScript(true)
  }, [])

  if (!accountId) return null
  return (
    <div>
      <div className="be-ix-link-block" />
      {shouldLoadScript ? <BrightEdgeScript accountId={accountId} /> : null}
    </div>
  )
}

export default BrightEdge
