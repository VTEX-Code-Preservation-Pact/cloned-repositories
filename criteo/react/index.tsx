import { canUseDOM } from 'vtex.render-runtime'
import { PixelMessage } from './typings/events'
import { fetchEmail, getSiteType } from './helpers'

async function dispatchEvent(event: CriteoQ) {
  const { criteo_q = [], criteo_id: account } = window
  const setAccount: CriteoAccountEvent = { event: 'setAccount', account }
  const setEmail: CriteoEmailEvent = {
    event: 'setEmail',
    email: [await fetchEmail()],
  }
  const setSiteType: CriteoSiteTypeEvent = {
    event: 'setSiteType',
    type: getSiteType(),
  }
  criteo_q.push(setAccount, setEmail, setSiteType, event)
}

function handleMessages(event: PixelMessage) {
  const { criteo_q = [], criteo_id: account } = window
  if (!account) return

  switch (event.data.eventName) {
    case 'vtex:pageInfo': {
      if (event.data.eventType === 'homeView') {
        const setHomeView: CriteoViewHomeEvent = {
          event: 'viewHome',
          tms: 'gtm-vtex',
        }
        dispatchEvent(setHomeView)
      }
      break
    }
    case 'vtex:departmentView':
    case 'vtex:internalSiteSearchView':
    case 'vtex:categoryView': {
      const {
        data: { products },
      } = event
      const item: string[] = products
        .slice(0, 3)
        .map<string>(({ productId }) => productId)
      const setViewList: CriteoViewListEvent = {
        event: 'viewList',
        tms: 'gtm-vtex',
        item,
      }
      dispatchEvent(setViewList)
      break
    }
    case 'vtex:productView': {
      const {
        data: {
          product: { productId },
        },
      } = event
      const setViewItem: CriteoViewItemEvent = {
        event: 'viewItem',
        tms: 'gtm-vtex',
        item: productId,
      }
      dispatchEvent(setViewItem)
      break
    }
    case 'vtex:orderPlaced': {
      const {
        data: { transactionId, transactionProducts },
      } = event

      const item: CriteoTrackTransactionItem[] = transactionProducts.map<
        CriteoTrackTransactionItem
      >(({ id, sellingPrice, quantity }) => ({
        id,
        price: sellingPrice,
        quantity,
      }))

      const setTrackTransaction: CriteoTrackTransactionEvent = {
        event: 'trackTransaction',
        id: transactionId,
        tms: 'gtm-vtex',
        item,
      }

      dispatchEvent(setTrackTransaction)
      break
    }
    default:
      break
  }
}

if (canUseDOM) {
  window.addEventListener('message', handleMessages)
}
