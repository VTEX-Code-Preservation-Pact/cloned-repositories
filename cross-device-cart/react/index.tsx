import { SessionWrapper as CrossCartDeviceBlock } from './components/SessionWrapper'

export default CrossCartDeviceBlock
