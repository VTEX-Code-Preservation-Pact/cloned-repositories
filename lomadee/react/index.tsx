import { getProductPrice, getProductAvable } from "./helpers/index";
import { PixelMessage } from "./typings/events";
import { canUseDOM } from "vtex.render-runtime";

function purchaseInjection(advertiserId: string, lomadeeEventId: string, transactionId: string, transactionTotalWithoutFreight: string) {
  const img = document.createElement("img");
  const url = `https://secure.lomadee.com/at/actionlog?adv=${advertiserId}&country=BR&transaction=${transactionId}&event1=${lomadeeEventId}&value1=${transactionTotalWithoutFreight}`;
  img.setAttribute("id", "lomadee-pixel");
  img.setAttribute("src", url);
  document.body.appendChild(img);
}

function handleMessages(e: PixelMessage) {
  let lomadee_datalayer: any;

  switch (e.data.eventName) {
    case "vtex:pageInfo": {
      lomadee_datalayer = {
        page: "home"
      };
      break;
    }

    case "vtex:orderPlaced": {
      lomadee_datalayer = {
        page: "conversion",
        conversion: {
          transactionId: e.data.transactionId,
          amount: e.data.transactionSubtotal,
          currency: "BRL",
          paymentType: e.data.transactionPaymentType[0].paymentSystemName,
          items: e.data.transactionProducts
        }
      };

      purchaseInjection(window.advertiserId, window.lomadeeEventId, e.data.transactionId, e.data.transactionSubtotal);
      break;
    }

    case "vtex:productView": {
      const {
        product: { productId, productName }
      } = e.data;
      lomadee_datalayer = {
        page: "product",
        product: {
          sku: productId,
          name: productName,
          price: getProductPrice(e.data.product),
          available: getProductAvable(e.data.product)
        }
      };
      break;
    }

    case "vtex:addToCart": {
      lomadee_datalayer = {
        page: "cart",
        cart: {
          skus: e.data.items[0].skuId
        }
      };

      break;
    }

    case "vtex:internalSiteSearchView": {
      const category = e.data.products[0].categories[0].split("/");
      lomadee_datalayer = {
        page: "search",
        keyword: category[2]
      };
      break;
    }

    case "vtex:categoryView": {
      const category = e.data.products[0].categories[0].split("/");
      lomadee_datalayer = {
        page: "category",
        category: {
          name: category[2]
        }
      };
      break;
    }

    default: {
      lomadee_datalayer = {
        page: "visit"
      };
      break;
    }
  }
}

if (canUseDOM) {
  window.addEventListener("message", handleMessages);
}
