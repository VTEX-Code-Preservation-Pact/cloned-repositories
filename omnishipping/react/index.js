import React from 'react'
import { Provider } from 'react-redux'
import OrderFormContainer from './containers/OrderFormContainer'
import CheckoutContainer from './containers/CheckoutContainer'
import configureStore from './store/configureStore'
import setPolyfills from './utils/Polyfills'
import RulesContainer from './containers/RulesContainer'
import { NoSSR } from 'render'
import WaitingState from './components/WaitingState'

window.omnishipping = {
  version: process.env.REACT_APP_VERSION,
}

const OmniShipping = () => {
  if (typeof document === 'undefined' || typeof document === 'undefined') {
    return null
  }

  setPolyfills()

  return (
    <NoSSR onSSR={<WaitingState loading />}>
      <Provider store={configureStore}>
        <RulesContainer
          render={() => (
            <CheckoutContainer render={() => <OrderFormContainer />} />
          )}
        />
      </Provider>
    </NoSSR>
  )
}

export default OmniShipping
