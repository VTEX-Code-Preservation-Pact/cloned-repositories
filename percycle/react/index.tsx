import { PixelMessage } from "./typings/events";
import { canUseDOM } from "vtex.render-runtime";

function handleMessages(e: PixelMessage) {
  const { storeName } = window

  if (e.data.eventName === 'vtex:orderPlaced') {
    const percycleScript = document.createElement('script');
    percycleScript.setAttribute('src', `//static.percycle.com/${storeName}/c.js`);
    percycleScript.setAttribute('async', 'async');
    percycleScript.setAttribute('defer', 'defer');
    document.getElementsByTagName('body')[0].appendChild(percycleScript);
  }
}

if (canUseDOM) {
  window.addEventListener("message", handleMessages);
}
