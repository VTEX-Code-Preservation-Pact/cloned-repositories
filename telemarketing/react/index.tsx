import React, { FC, Suspense } from 'react'
import { NoSSR } from 'vtex.render-runtime'
import isMyVtex from './utils/isMyVtex'

const TelemarketingContainer = React.lazy(() => import('./components/TelemarketingContainer'))

const Telemarketing: FC = (props) => {
  if (!isMyVtex()) {
    return null
  }

  return (
    <NoSSR>
      <Suspense fallback={<React.Fragment />}>
        <TelemarketingContainer {...props} />
      </Suspense>
    </NoSSR>
  )
}

export default Telemarketing
