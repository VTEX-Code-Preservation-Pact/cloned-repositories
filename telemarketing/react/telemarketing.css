.wrapper {}

.container .popoverArrowUp {
  right: 20px;
}

.container .popoverBox {
  min-width: 320px;
}

/* Popover css */

.popoverArrowUp {
  width: 25px;
  height: 25px;
  left: 86%;
}

.popoverBox {
  visibility: visible;
  min-width: 310px;
}

.popoverContentContainer {
  min-width: 100%;
}

.clientName {
  word-break: break-word;
}

.clientNameBar {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 90%;
  text-align: left;
}

@media screen and (aspect-ratio: 40/71) {
  .popoverBox {
    right: -2.4vw;
    width: 100vw;
    top: 4.1vh !important;
  }
}

.login {
}

.popoverHeaderIcon {
}

.popoverHeaderEmail {
}

.loginForm {
}

.loginFormMessage {
}

.emailInput {
}

.logout {
}

.logoutForm {
}

.popoverContainer {
}
