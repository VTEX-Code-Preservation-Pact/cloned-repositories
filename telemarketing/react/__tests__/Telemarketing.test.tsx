import React from 'react'
import { mergeDeepRight } from 'ramda'
import { render, fireEvent } from '@vtex/test-tools/react'

import Telemarketing from '../components/Telemarketing'

describe('<Telemarketing /> component', () => {
  const renderComponent = (customProps = {}) => {
    const defaultProps = {
      attendantEmail: 'attendant@vtex.com',
      emailInput: 'email@vtex.com',
      loading: false,
      onDepersonify: () => {},
      onInputChange: () => {},
      onImpersonate: () => {},
    }

    return render(<Telemarketing {...defaultProps} {...customProps} />)
  }

  it('should match snapshot without client', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should match snapshot with client', () => {
    const client: Client = {
      document: 'document',
      phone: 'phone',
      name: 'name',
      email: 'client@vtex.com',
    }

    const { asFragment } = renderComponent({ client: client })

    expect(asFragment()).toMatchSnapshot()
  })

  it('should show attendant email', () => {
    const email = 'nice_attendant_email@vtex.com'
    const { getByText } = renderComponent({ attendantEmail: email })

    const attendantEmail = getByText(email)
    expect(attendantEmail).toBeTruthy()
  })

  it('should show login button and form', () => {
    const { getByText, getByPlaceholderText } = renderComponent()

    const emailPlaceholder = getByPlaceholderText('Ex: example@mail.com')
    expect(emailPlaceholder).toBeTruthy()

    const login = getByText('Login')
    expect(login).toBeTruthy()
  })

  it('should show client data', () => {
    const client: Client = {
      document: 'my client document',
      phone: '+9999999999999',
      name: 'my client name',
      email: 'client@vtex.com',
    }

    const { getByText, getAllByText } = renderComponent({ client: client })

    const document = getByText(client.document)
    expect(document).toBeTruthy()

    const phone = getByText(client.phone)
    expect(phone).toBeTruthy()

    const name = getAllByText(client.name)
    expect(name).toBeTruthy()

    const email = getByText(client.email)
    expect(email).toBeTruthy()
  })

  it('should login when clicked', () => {
    const onImpersonate = jest.fn()
    const { getByText } = renderComponent({ onImpersonate: onImpersonate })

    fireEvent.click(getByText('Login'))

    expect(onImpersonate).toBeCalledTimes(1)
  })

  it('should logout when clicked', () => {
    const onDepersonify = jest.fn()
    const client: Client = {
      document: 'my client document',
      phone: '+9999999999999',
      name: 'my client name',
      email: 'client@vtex.com',
    }

    const { getByText } = renderComponent({
      client: client,
      onDepersonify: onDepersonify,
    })

    fireEvent.click(getByText('Logout'))

    expect(onDepersonify).toBeCalledTimes(1)
  })
})
