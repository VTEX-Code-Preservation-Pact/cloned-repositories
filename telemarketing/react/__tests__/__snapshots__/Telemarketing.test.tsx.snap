// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Telemarketing /> component should match snapshot with client 1`] = `
<DocumentFragment>
  <div
    class="w-100 wrapper bg-emphasis c-on-emphasis"
  >
    <section
      class="Container-mock container flex justify-center tc h2 t-mini pa2 w-100"
    >
      <div
        class="telemarketingBar flex justify-between w-100 mw9"
      >
        <div
          class="attendantContainer flex pl3 w-50 items-center c-on-base--inverted"
        >
          <svg
            class="IconAssistantSales-mock  "
            height="20"
            width="20"
          >
            <rect
              height="20"
              width="20"
            />
          </svg>
          <div
            class="attendantEmail ml2"
          >
            <span>
              Attendant: attendant@vtex.com
            </span>
          </div>
        </div>
        <div
          class="logout false"
        >
          <div
            class="popoverContainer relative flex h-100 items-center pr4"
          >
            <div
              class="pointer w-100"
            >
              <div
                class="logoutHeader c-on-base--inverted flex items-center"
              >
                <svg
                  class="IconProfile-mock  "
                  height="20"
                  width="20"
                >
                  <rect
                    height="20"
                    width="20"
                  />
                </svg>
                <div
                  class="pa2 clientNameBar w-100"
                >
                  name
                </div>
              </div>
            </div>
            <div>
              <div
                class="popoverBox absolute-ns top-2 z-max bb b--muted-3 fixed-s dn"
              >
                <div
                  class="popoverContentContainer mt3-ns bg-base shadow-3-ns"
                >
                  <div
                    class="popoverHeader bg-emphasis w-100 pa7"
                  >
                    <div
                      class="popoverHeaderIcon c-on-base--inverted"
                    >
                      <svg
                        class="IconAssistantSales-mock  "
                        height="50"
                        width="50"
                      >
                        <rect
                          height="50"
                          width="50"
                        />
                      </svg>
                    </div>
                    <div
                      class="popoverHeaderEmail c-on-emphasis"
                    >
                      attendant@vtex.com
                    </div>
                  </div>
                  <div
                    class="logoutInfoContainer bg-base w-100 pt7 pb5 ph5"
                  >
                    <div
                      class="logoutForm c-disabled"
                    >
                      <div
                        class="w-100 bw1 bb b--muted-5 flex-wrap"
                      >
                        <div
                          class="clientName w-100 t-heading-5 center pb5 c-on-base"
                        >
                          name
                        </div>
                        <div
                          class="emailContainer w-100 flex flex-wrap t-small"
                        >
                          <div
                            class="emailField tl pb5 pr2"
                          >
                            Email
                          </div>
                          <div
                            class="emailValue pb5 pl2 c-muted-1"
                          >
                            client@vtex.com
                          </div>
                        </div>
                        <div
                          class="documentContainer w-100 flex flex-wrap t-small"
                        >
                          <div
                            class="documentField tl pb5 pr2"
                          >
                            <span>
                              Document
                            </span>
                          </div>
                          <div
                            class="documentValue pb5 pl2 c-muted-1"
                          >
                            document
                          </div>
                        </div>
                        <div
                          class="phoneContainer w-100 flex flex-wrap t-small"
                        >
                          <div
                            class="phoneField tl pb5 pr2"
                          >
                            <span>
                              Phone
                            </span>
                          </div>
                          <div
                            class="phoneValue pb5 pl2 c-muted-1"
                          >
                            phone
                          </div>
                        </div>
                      </div>
                      <div
                        class="logoutButtonsContainer flex justify-around mt5"
                      >
                        <a
                          href="store.account"
                        >
                          <button
                            type="button"
                          >
                            <span>
                              Orders
                            </span>
                          </button>
                        </a>
                        <button
                          type="button"
                        >
                          <span>
                            Logout
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="popoverArrowUp absolute top-0 rotate-135 dib-ns dn-s bg-emphasis"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`<Telemarketing /> component should match snapshot without client 1`] = `
<DocumentFragment>
  <div
    class="w-100 wrapper bg-base--inverted c-on-base--inverted"
  >
    <section
      class="Container-mock container flex justify-center tc h2 t-mini pa2 w-100"
    >
      <div
        class="telemarketingBar flex justify-between w-100 mw9"
      >
        <div
          class="attendantContainer flex pl3 w-50 items-center c-on-base--inverted"
        >
          <svg
            class="IconAssistantSales-mock  "
            height="20"
            width="20"
          >
            <rect
              height="20"
              width="20"
            />
          </svg>
          <div
            class="attendantEmail ml2"
          >
            <span>
              Attendant: attendant@vtex.com
            </span>
          </div>
        </div>
        <div
          class="login w-50 flex flex-row-reverse-ns flex-row-s"
        >
          <div
            class="popoverContainer relative flex h-100 items-center pr4"
          >
            <div
              class="pointer w-100"
            >
              <div
                class="loginButton flex items-center c-on-base--inverted"
              >
                <svg
                  class="IconProfile-mock  "
                  height="20"
                  width="20"
                >
                  <rect
                    height="20"
                    width="20"
                  />
                </svg>
                <div
                  class="loginAsText ml2"
                >
                  <span>
                    Login as
                  </span>
                </div>
              </div>
            </div>
            <div>
              <div
                class="popoverBox absolute-ns top-2 z-max bb b--muted-3 fixed-s dn"
              >
                <div
                  class="popoverContentContainer mt3-ns bg-base shadow-3-ns"
                >
                  <div
                    class="popoverHeader bg-base--inverted w-100 pa7 c-on-base--inverted"
                  >
                    <div
                      class="popoverHeaderIcon"
                    >
                      <svg
                        class="IconAssistantSales-mock  "
                        height="50"
                        width="50"
                      >
                        <rect
                          height="50"
                          width="50"
                        />
                      </svg>
                    </div>
                    <div
                      class="popoverHeaderEmail white-50 mt3"
                    >
                      attendant@vtex.com
                    </div>
                  </div>
                  <div
                    class="loginFormContainer bg-base w-100 ph5 pb5 pt7"
                  >
                    <div
                      class="loginForm c-disabled"
                    >
                      <div
                        class="loginFormMessage t-small tl mb3"
                      >
                        <span>
                          Login as
                        </span>
                      </div>
                      <div
                        class="emailInput mb5"
                      >
                        <input
                          placeholder="Ex: example@mail.com"
                          value="email@vtex.com"
                        />
                      </div>
                      <button
                        type="button"
                      >
                        <span>
                          Login
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="popoverArrowUp absolute top-0 rotate-135 dib-ns dn-s bg-base--inverted"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
